client/starwhale/core/dataset/store.py

```diff
@@ -408,6 +415,30 @@ def _make_file(self, bucket: str, key_compose: str) -> FileLikeObj:
             return io.BytesIO(f.read(_end - _start + 1))
 
 
+class SignedUrlBackend(StorageBackend, CloudRequestMixed):
+    def __init__(self, dataset_uri: URI) -> None:
+        super().__init__(kind=SWDSBackendType.SignedUrl)
+        self.dataset_uri = dataset_uri
+
+    def _make_file(self, auth: str, key_compose: t.Tuple[str, int, int]) -> FileLikeObj:
+        _key, _start, _end = key_compose
+        url = self.sign_uri(_key, auth)
+        headers = {}
+        if _start and _end:
```

Comment:
if only set start or end, range does not work?

Reply:
yes, both should be set